Cinema-Reservation-System-Front-End-Client-Part-Internship-project/src/app/shared/http-error.interceptor.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, catchError, throwError } from 'rxjs';
+import { ToastService } from '../toasts/toast.service';
+import { ErrorObj } from './error-obj.model';
+
+@Injectable()
+export class HttpErrorInterceptor implements HttpInterceptor {
+  constructor(private toastService: ToastService) {}
+
+  intercept(
+    request: HttpRequest<unknown>,
+    next: HttpHandler,
+  ): Observable<HttpEvent<unknown>> {
+    console.log('ERROR INTERCEPTOR CALLED');
+
+    return next.handle(request).pipe(
+      catchError((err: HttpErrorResponse) => {
```

Comment:
Let's add here a condition where it is an Unauthorized error, we will redirect user to the Login page